### PR TITLE
Fix ROIs in model generation; fixes #604 and #602

### LINKED ISSUE #604: Model creation crashes when there are multiple ROI
Working with stis1.ecsv, I defined 4 ROIs then tried to create a 2nd-order polynomial model. SpecViz crashed with the following traceback.

```
Traceback (most recent call last):
  File "/Users/hcferguson/anaconda3/envs/specutils_dev/lib/python3.6/site-packages/specviz-0.6.dev1898-py3.6.egg/specviz/plugins/model_editor/model_editor.py", line 68, in <lambda>
    action.triggered.connect(lambda x, m=v: self._add_fittable_model(m))
  File "/Users/hcferguson/anaconda3/envs/specutils_dev/lib/python3.6/site-packages/specviz-0.6.dev1898-py3.6.egg/specviz/plugins/model_editor/model_editor.py", line 263, in _add_fittable_model
    model = initialize(model, spec.spectral_axis, spec.flux)
AttributeError: 'list' object has no attribute 'spectral_axis'
[1]    97936 abort      specviz
```

### PR DESCRIPTION
**Pull request by nmearl**
FIxes #604 and #602.

### CODE REVIEW

File: specviz/plugins/model_editor/model_editor.py [hunk 1]
```diff
@@ -259,7 +259,10 @@ def _add_fittable_model(self, model_type):
         spec = self._get_selected_plot_data_item().data_item.spectrum
 
         if inc_regs is not None:
-            spec = extract_region(spec, inc_regs)
+            exc_regs = inc_regs.invert_from_spectrum(spec) \
+                if inc_regs is not None else None
```
Comment (brechmos-stsci) [inline specviz/plugins/model_editor/model_editor.py]: For consistency this seems to be the same structure as above change, so how about just:
```
       exc_regs = inc_regs.invert_from_spectrum(spec) \
            if inc_regs is not None else None
```
though maybe I am missing something.

Comment (brechmos-stsci) [reply specviz/plugins/model_editor/model_editor.py]: nvm, missed the line below.

**Comment by brechmos-stsci** [comment]
I think I need a bit of guidance here. I am not sure how to setup in order to get to that bit of code. I loaded a 1D data file, created two regions (hoping to use those to exclude for model fitting), then created a 2nd order polynomial. The fitting works (but it is a bad fit). How do I specify that these two regions I created should be used to exclude?  

Or quite possibly I am missing something with what @hcferguson  was doing in https://github.com/spacetelescope/specviz/issues/604 and what this fix is for.

This does appear to fix https://github.com/spacetelescope/specviz/issues/602.

**Comment by nmearl** [comment]
The ROIs represent regions to be included. The code retrieves the current ROI select (`inc_regs`), and then inverts them to get the non-ROI selection of the spectrum (`exc_regs`), and feeds that into the specutils utilities (since they take excluded regions, not included regions). Hope that makes sense!

**Comment by brechmos-stsci** [comment]
Cool. Seems to work for me, then.

**Pull request merged**